microcloud/go.mod

```diff
@@ -1,20 +1,20 @@
 module github.com/canonical/microcloud/microcloud
 
-go 1.22
+go 1.22.0
```

Comment:
Not a blocker, just wanted to know why we update to `1.22.0` here?

Reply:
When enabling CodeQL, I noticed it would choke on that short `1.22` version string: https://github.com/canonical/microcloud/actions/runs/8576568978/job/23507751328#step:4:90

```sh
Attempting to automatically build go code
  /opt/hostedtoolcache/CodeQL/2.17.0/x64/codeql/go/tools/autobuild.sh 
  2024/04/05 22:48:49 Autobuilder was built with go1.22.1, environment has go1.21.8
  2024/04/05 22:48:49 LGTM_SRC is /home/runner/work/microcloud/microcloud
  2024/04/05 22:48:49 Found no go.work files in the workspace; looking for go.mod files...
  2024/04/05 22:48:49 Found 1 go.mod file(s).
  2024/04/05 22:48:49 Import path is 'github.com/canonical/microcloud'
  go: downloading go1.22 (linux/amd64)
  go: download go1.22 for linux/amd64: toolchain not available
```

Just adding that `.0` made it happy.

Reply:
Ah ok, thanks for the hint!